### PR TITLE
Include steps for getting workspaces logs in end-user guide

### LINKED ISSUE #17459: Getting workspaces logs should be in end-user guide
### Is your enhancement related to a problem? Please describe.

Page [viewing Che workspaces logs](https://www.eclipse.org/che/docs/che-7/viewing-che-workspaces-logs/) is currently in the Che Admin Guide. But those infos are gold for an end-user that tries to troubleshoot a workspace that is failing to start. And he will never find those in the section dedicated to him.

### Describe the solution you'd like

Move a that page in the end-user guide. Probably in [troubleshooting for che end users](https://www.eclipse.org/che/docs/che-7/troubleshooting-for-che-end-users/https://www.eclipse.org/che/docs/che-7/troubleshooting-for-che-end-users/) section.

Mention a link to that page from the admin guide (just in case).

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Adds link to viewing workspace logs to the end-user troubleshooting guide. To make this a tidy change, the end-user troubleshooting section is split into subtopics: 
1. Viewing workspace logs
2. Starting a workspace in debug mode
3. Switching workspace storage types

I also went and fixed a few broken links, and did a quick edit pass on the existing troubleshooting guide.

Each commit in this PR should have a fairly self-explanatory message; feel free to ask if it is unclear.

### What issues does this PR fix or reference?
Fixes https://github.com/eclipse/che/issues/17459

### Specify the version of the product this PR applies to. 
\>=7.17.x

### CODE REVIEW

**Comment by amisevsk** [comment]
I've run linkchecker and it only reports one broken link (which I think is a false positive as it contains `{prod-short}`)

Running 
```
for file in $(git diff master --numstat | grep adoc | cut -f 3); do 
    bash tools/test-adoc.sh $file | grep -E 'FAIL|$'
done
```
on this branch gives one new failure:
```
Testing file: /home/amisevsk/git/che-docs/src/main/pages/che-7/end-user-guide/assembly_troubleshooting-for-che-end-users.adoc

  Document type: assembly

  [ FAIL ]  Assembly does not contain any included modules.

Checked 6 item(s) in 1 file(s), found 1 problem(s).
```
I don't know how to resolve this; all of the "Overview" sections I can see have the same issue.

**Comment by amisevsk** [comment]
This PR also fixes a few minor issues in the existing end-user troubleshooting docs, such as [stray front matter](https://www.eclipse.org/che/docs/che-7/troubleshooting-for-che-end-users/#using-diffrent-type-of-storage_troubleshooting-for-che-end-users) included in the storage types procedure doc

**Comment by amisevsk** [comment]
@rkratky One note about this PR and downstreaming: in the new overview for end-user troubleshooting, I'm linking to an assembly in the admin guide. Please let me know if this case needs to be handled specially so as to not break the link.

**Comment by amisevsk** [comment]
Rebased onto master and adapted to recent changes for storage types.

**Comment by amisevsk** [comment]
cc: @olexii4 this PR turns the End User tutorial into a folder with subassemblies; this may be useful for https://github.com/eclipse/che-docs/pull/1438

File: src/main/pages/che-7/installation-guide/examples/system-variables.adoc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-[id="che-server"]
+[id="LChe-server"]
```
Comment (amisevsk) [inline src/main/pages/che-7/installation-guide/examples/system-variables.adoc:1]: No idea if this commit should be dropped, I was following the contributing guide and had some changes to the repo.

**Comment by rkratky** [comment]
Implemented through https://github.com/eclipse/che-docs/pull/1495 and other PRs. Thanks, @amisevsk.

**Comment by amisevsk** [comment]
@rkratky Looking at the current Che docs, it looks like the original issue (https://github.com/eclipse/che/issues/17459) has not been resolved -- getting workspaces logs is still in the admin guide.

**Pull request closed without merging**